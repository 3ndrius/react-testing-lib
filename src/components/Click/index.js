import React from 'react'

export default function Click() {
    const [ valueClick, setValueClick ] = React.useState(0)
    return (
        <div className="click">
            <button>Up</button>
            <button>Down</button>
            <h2 data-testid="counter-value">{valueClick}</h2>
        </div>
    )
}
