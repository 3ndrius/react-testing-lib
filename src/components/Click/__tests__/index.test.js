import React from 'react';
import { render, cleanup } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect'
import Click from '../index'

afterEach(cleanup);

describe('render click component', () => {

    it('render component click', () => {
        const { asFragment } = render(<Click />);
        expect(asFragment).toMatchSnapshot();
    })

    it('display count value ', () => {
        const { getByTestId } = render(<Click />);
        expect(getByTestId('counter-value')).toHaveTextContent('0');
    })
})