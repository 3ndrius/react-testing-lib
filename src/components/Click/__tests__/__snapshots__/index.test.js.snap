// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render click component render component click 1`] = `[Function]`;
